source/opt/block_merge_util.cpp

```diff
@@ -173,7 +173,9 @@ void MergeWithSuccessor(IRContext* context, Function* func,
       auto& vec = terminator->dbg_line_insts();
       auto& new_vec = merge_inst->dbg_line_insts();
       new_vec.insert(new_vec.end(), vec.begin(), vec.end());
-      terminator->clear_dbg_line_insts();
+      terminator->ClearDbgLineInsts();
+      for (auto& l_inst : new_vec)
+        context->get_def_use_mgr()->AnalyzeInstDefUse(&l_inst);
```

Comment:
The act of copying line Instructions from terminator to merge_inst creates new Instruction objects. That is because the dbg_line_insts_ in Instruction is a vector of Instruction, NOT a vector of unique pointers to Instructions like the rest of the Instruction vectors we deal with. The DefUse manager contains pointers to Instructions, so when we create new line Instructions in merge_inst, we have do AnalyzeInstDefUse() on them. Technically, we don't have to do AnalyzeInstDefUse() on ALL the members of new_vec, only the ones added from terminator. But doing ALL of new_vec was simpler and less error prone, if slightly less efficient. I can fix this to only do the analysis that is truly needed if you prefer.

Reply:
Ah .. sorry. I was confused. I expect `new_vec.size()` in the line 174 is not that big, so this looks fine to me.